src/Workspaces/Remote/ServiceHub/Services/AssetSynchronization/RemoteAssetSynchronizationService.cs

```diff
@@ -65,7 +65,7 @@ public ValueTask SynchronizeTextAsync(DocumentId documentId, Checksum baseTextCh
                     // Now attempt to manually apply the edit, producing the new forked text.  Store that directly in
                     // the asset cache so that future calls to retrieve it can do so quickly, without synchronizing over
                     // the entire document.
-                    var newText = new SerializableSourceText(text.WithChanges(textChanges));
+                    var newText = new SerializableSourceText(text.WithChanges(textChanges), text.GetContentHash());
                     var newChecksum = serializer.CreateChecksum(newText, cancellationToken);
```

Comment:
this is not any extra expense.  The call to serializer.CreateChecksum will force the content hash to be created anyways.